### DISCUSSION TITLE
UserWarning: cleaning up ddp environment...

### DISCUSSION THREAD

**Discussion by tryumanshow** [Lightning-AI/pytorch-lightning#7820] [Lightning Trainer API: Trainer, LightningModule, LightningDataModule] [answered]
When I execute some program in PyTorch-Lightning, it implements very fast. But, at the last part, I got a message as below

```
/home/mydirectory/anaconda3/lib/python3.7/site-packages/pytorch_lightning/utilities/distributed.py:69: UserWarning: cleaning up ddp environment...

warnings.warn(*args, **kwargs)
```

After this message, all the system stops. I can't see the result that I want stopped by this message.

Can anyone tell me how to solve it? I'm in Hurry ㅠ.ㅠ

(I am using it with `wandb` & `HuggingFace`.)

**Comment by Borda** [comment, marked as answer]
@data-weirdo mind share some sample code to reproduce? I have been using DDP in some of our examples and all is fine :rabbit:

**Reply by tryumanshow** [reply]
Thank you for replying on my question. 

I found the reason why I got this problem. It was my very first time to use PyTorchLightning on training sth, and I was using 'prepare_data' method inherited from LightningDataModule class. If I stop using 'prepare_data', then it runs without any problem. I didn't explore the reason in depth, but anyway it helped me. :)
